gammapy/spectrum/models.py

```diff
@@ -1464,3 +1465,130 @@ def evaluate(self, energy, **kwargs):
+    likelihood spectral fit of the model's parameters directly on observations, as opposed to the MCMC
+    `fit to flux points <https://naima.readthedocs.io/en/latest/mcmc.html>`_ featured in
+    Naima. All the parameters defining the parent population of charged particles are stored as
+    `~gammapy.utils.modeling.Parameter` and left free by default.
+    In case that the radiative model is `~naima.radiative.Synchrotron`,
+    the magnetic field strength may also be fitted. Parameters can be freezed/unfreezed before the fit, and
+    maximum/minimum values can be set to limit the parameters space to the physically interesting region
+    (see the example below).
+
+    Parameters
+    ----------
+    radiative_model : `~naima.models.BaseRadiative`
+        An instance of a radiative model defined in `~naima.models`
+    distance : `~astropy.units.Quantity`, optional
+        Distance to the source. If set to 0, the intrinsic differential
+        luminosity will be returned. Default is 1 kpc
+    seed : str or list of str, optional
+        Seed photon field(s) to be considered for the `radiative_model` flux computation,
+        in case of a `~naima.models.InverseCompton` model. It can be a subset of the
+        `seed_photon_fields` list defining the `radiative_model`. Default is the whole list
+        of photon fields
+
+    Examples
+    --------
+    Create and plot a spectral model that convolves an `ExponentialCutoffPowerLaw` electron distribution
+    with an `InverseCompton` radiative model, in the presence of multiple seed photon fields.
+
+    .. plot::
+        :include-source:
+
+        import naima
+        from gammapy.spectrum.models import NaimaModel
+        import astropy.units as u
+        import matplotlib.pyplot as plt
+
+
+        particle_distribution = naima.models.ExponentialCutoffPowerLaw(1e30 / u.eV, 10 * u.TeV, 3.0, 30 * u.TeV)
+        radiative_model = naima.radiative.InverseCompton(
+            particle_distribution,
+            seed_photon_fields=[
+                "CMB",
+                ["FIR", 26.5 * u.K, 0.415 * u.eV / u.cm ** 3],
+            ],
+            Eemin=100 * u.GeV,
+        )
+
+        model = NaimaModel(radiative_model, distance=1.5 * u.kpc)
+
+        opts = {
+            "energy_range" : [10 * u.GeV, 80 * u.TeV],
+            "energy_power" : 2,
+            "flux_unit" : "erg-1 cm-2 s-1",
+        }
+
+        # Plot the total inverse Compton emission
+        model.plot(label='IC (total)', **opts)
+
+        # Plot the separate contributions from each seed photon field
+        for seed, ls in zip(['CMB','FIR'], ['-','--']):
+            model = NaimaModel(radiative_model, seed=seed, distance=1.5 * u.kpc)
+            model.plot(label="IC ({})".format(seed), ls=ls, color="gray", **opts)
+
+        plt.legend(loc='best')
+        plt.show()
+    """
+    #TODO: prevent users from setting new attributes after init
```

Comment:
Remove TODO? Or is this really something we want / need to do here?

Reply:
All spectral models prevent the intialization of new attributes after `__init__` because they have parameters declared in `__slots__`. For Naima models this turns out to be tricky (see previous answer: https://github.com/gammapy/gammapy/pull/2124#discussion_r277664181), but for consistency I think we will probably need to implement this TODO here